### PR TITLE
deploy: generate separate crds.yaml, rbac.yaml & setup-controller.yaml

### PR DESCRIPTION
**Pull request by Rakshith-R**
This commit adds capability to create separate crds.yaml, rbac.yaml and setup-controller.yaml.
This will reduce number of lines in each file, increase readability, customizability and yamls
required for deployment will be on par with other projects.
This also adds resource limits to rbac-proxy container, and moves namespace to config/rbac.

Signed-off-by: Rakshith R <rar@redhat.com>

### CODE REVIEW

File: config/rbac/namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    control-plane: controller-manager
+  name: system
```
Comment (Madhu-1) [inline config/rbac/namespace.yaml:6]: csi-addons should gets deployed in csi-addons-system not in system namespace?

**Review by Madhu-1 — COMMENTED** [review]
left questions, i don't have major concerns

File: config/rbac/namespace.yaml (hunk 1 above)
Comment (Rakshith-R) [reply config/rbac/namespace.yaml:6]: > csi-addons should gets deployed in csi-addons-system not in system namespace?

`csi-addons-` will get prefixed from kustomize settings.

File: config/manager/manager_auth_proxy_patch.yaml [hunk 2]
```diff
@@ -20,6 +20,15 @@ spec:
...
           name: https
+        # TODO(user): Configure the resources accordingly based on the project requirements.
+        # More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+        resources:
+          limits:
+            cpu: 500m
+            memory: 128Mi
+          requests:
+            cpu: 10m
+            memory: 64Mi
```
[the remark this answers is not in the document]
Comment (Rakshith-R) [reply config/manager/manager_auth_proxy_patch.yaml:31]: > let's not add default from our side as we don't know the exact default.

this is the same as from controller, It has resource defaults and rbac-proxy container was missing it, so added them.

**Review by Madhu-1 — DISMISSED** [review]

File: config/manager/manager_auth_proxy_patch.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Rakshith-R) [reply config/manager/manager_auth_proxy_patch.yaml:31]: https://github.com/csi-addons/kubernetes-csi-addons/blob/57ed4f6211c0847eecead1d6f7baaee15f8e356d/deploy/controller/setup-controller.yaml#L1038

File: config/manager/manager.yaml [hunk 3]
```diff
@@ -56,5 +49,5 @@ spec:
           requests:
             cpu: 10m
             memory: 64Mi
-      serviceAccountName: controller-manager
+      serviceAccountName: csi-addons-controller-manager
```
Comment (nixpanic) [inline config/manager/manager.yaml:52]: Why is this not taken from the default anymore?

**Review by nixpanic — CHANGES_REQUESTED** [review]
Please split this PR in separate commits, maybe even separate PRs.

The creation of separate files seems independent of lots of other changes. This makes reviewing more confusing than it should.

File: config/manager/manager.yaml (hunk 3 above)
Comment (Rakshith-R) [reply config/manager/manager.yaml:52]: > Why is this not taken from the default anymore?

kustomize does not seem to add prefix to `serviceAccountName` when built from config/manager/kustomize.yaml

**Comment by Rakshith-R** [comment]
Added multiple small commits, PTAL @nixpanic @Madhu-1

**Review by nixpanic — APPROVED** [review]

**Comment by Rakshith-R** [comment]
@mergifyio rebase

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**